### PR TITLE
UHM-6465  Better progress indicator for printing lab results

### PR DESCRIPTION
**Pull request by brandones**
https://pihemr.atlassian.net/browse/UHM-6465

Lots of refactoring happened. I will note the actual changes in comments.

Peek 2022-05-06 17-37

### CODE REVIEW

File: app/css/lab-results-view.scss [hunk 1]
```diff
@@ -53,6 +53,7 @@
 .lab-results-list-cell-result {
   display: flex;
   justify-content: center;
+  text-align: center;
```
Comment (brandones) [inline app/css/lab-results-view.scss:56]: These CSS changes are new.

File: app/css/openmrs-owa-labworkflow.scss [hunk 2]
```diff
@@ -86,6 +86,11 @@ input.form-control:focus {
     margin: auto;
 }
 
+.big-spinner {
+    width: 4em;
+    height: 4em;
+}
```
Comment (brandones) [inline app/css/openmrs-owa-labworkflow.scss:92]: This is new

File: app/js/components/ConceptDisplay.jsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ const {
 } = formValidations;
 
 class ConceptDisplay extends PureComponent {
-  UNSAFE_componentWillMount() {
+  componentDidMount() {
```
Comment (brandones) [inline app/js/components/ConceptDisplay.jsx:15]: Just following React's migration guide.

File: app/js/components/LabResultsList/LabResultsList.jsx [hunk 4]
```diff
@@ -487,18 +254,15 @@ export class LabResultsList extends PureComponent {
       const labTestAndPanelTypes = getAllLabTestAndPanelTypes(labResults);
       return (
         <div className="main-container">
+          { isPrinting && <div className="spinner big-spinner" />}
```
Comment (brandones) [inline app/js/components/LabResultsList/LabResultsList.jsx:257]: This renders the new spinner.

File: app/js/components/LabResultsList/LabResultsList.jsx [hunk 5]
```diff
@@ -487,18 +254,15 @@ export class LabResultsList extends PureComponent {
...
           <ReactToPrint
             // eslint-disable-next-line
             trigger={() => (
               <button type="button" className="print-button" disabled={isPrinting}>
-                { isPrinting
```
Comment (brandones) [inline app/js/components/LabResultsList/LabResultsList.jsx:494]: This removes the old spinner from within the print button.

**Review by cioan — APPROVED** [review]
Looks good! Thanks @brandones !

**Pull request merged**